src/main/java/io/kestra/plugin/couchbase/Query.java

```diff
@@ -0,0 +1,136 @@
+package io.kestra.plugin.couchbase;
+
+import com.couchbase.client.java.Cluster;
+import com.couchbase.client.java.codec.TypeRef;
+import com.couchbase.client.java.json.JsonArray;
+import com.couchbase.client.java.json.JsonObject;
+import com.couchbase.client.java.query.QueryOptions;
+import com.couchbase.client.java.query.QueryResult;
+import io.kestra.core.models.annotations.Example;
+import io.kestra.core.models.annotations.Plugin;
+import io.kestra.core.models.tasks.RunnableTask;
+import io.kestra.core.runners.RunContext;
+import io.kestra.core.serializers.FileSerde;
+import io.kestra.plugin.couchbase.models.FetchType;
+import io.swagger.v3.oas.annotations.media.Schema;
+import lombok.*;
+import lombok.experimental.SuperBuilder;
+
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+import java.io.*;
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+@Schema(
+    title = "Query a Couchbase database with N1QL."
+)
+@Plugin(
+    examples = {
+        @Example(
+            title = "Send a N1QL query to a Couchbase Database",
+            code = {
+                "connectionString: couchbase://localhost",
+                "username: couchbase_user",
+                "password: couchbase_passwd",
+                "query: SELECT * FROM COUCHBASE_BUCKET(.COUCHBASE_SCOPE.COUCHBASE_COLLECTION)",
+                "fetchType: FETCH"
+            }
+        ),
+    }
+)
+public class Query extends CouchbaseConnection implements RunnableTask<Query.Output>, QueryInterface {
+    @NotNull
+    @Builder.Default
+    protected FetchType fetchType = FetchType.NONE;
+    protected Object parameters;
+
+    @NotNull
+    @NotBlank
+    protected String query;
+
+    public Output run(RunContext runContext) throws Exception {
+        Cluster session = connect(runContext);
+
+        String renderedQuery = runContext.render(query);
+        QueryOptions parametersForQuery = getParametersForQuery();
+        QueryResult result = session.query(renderedQuery, parametersForQuery);
+
+        close(session);
+
+        List<Map<String, Object>> rowsAsMap = result.rowsAs(new TypeRef<>() {
+        });
+
+        Output.OutputBuilder outputBuilder = Output.builder().size((long) rowsAsMap.size());
+        switch (fetchType) {
```

Comment:
You can use the new Java new switch expression syntax, see https://nipafx.dev/java-13-switch-expressions/

Reply:
It was still in Java 11 when I wrote this, gonna move on to it